### ISSUE TITLE
adjustment for softmax 

### ISSUE THREAD

**Issue by misbahch6** [kundajelab/deeplift#116]
Hello Avanti,

I am trying to understand the adjustment for softmax classes. 

For the following example, do we calculate mean contributions like this ?
Ref_input : {0,0,0,0}
Input: {0.5,0.3,1,1}

Contributions: {2,5,1,-1}

Ref_output: {0.2,0.8}
Ouptut: {0.9,0.1}

It is recommended to mean normalize the contributions. So should i just take the mean of contributions and minus it from the contributions?

Mean_normalized_contributions: {-1.5, 1.5, -2.5,-4.5 }

Or maybe i am completely wrong? and first i have to group all examples by classes and then rather calculate and subtract the mean feature wise for the that particular class ?

**Comment by AvantiShri** [comment]
It's more like this:

Original feature contributions to the class logits:
class 0: {1, 1, 0, -1}
class 1: {1, 0, 1, -3}

Mean feature contributions across logits of all classes: {1, 0.5, 0.5, -2}

Mean-normalized feature contributions:
class 0: {0, 0.5, -0.5, 1}
class 1: {0, -0.5, 0.5, -1}

Does that make sense? The goal is to account for the fact that if a feature contributes equally to the logits of all softmax classes, that means it effectively contributes to none of the softmax classes. For example, the first feature is contributing "1" to both class 0 and class 1; that means it is effectively contributing to none of the classes (because the contribution would cancel out the numerator and denominator of the softmax).

**Comment by misbahch6** [comment]
cystral clear now ... thank you so much ...
